applications/vanilla/library/class.categorycollection.php

```diff
@@ -461,6 +461,42 @@ public function getTree($parentID = -1, $options = []) {
         return $tree;
     }
 
+    /**
+     * Get the id's of a category's descendants.
+     *
+     * @param int $parentIDs
+     * @param array $options
+     * @return array
+     */
+    public function getDescendantIDs($parentIDs = -1, $options = []) {
+        $ids = [];
+        $parentIDs = [$parentIDs] ?: [-1];
+        $defaultOptions = [
+            'maxDepth' => 3,
+            'permission' => 'PermsDiscussionsView'
+        ];
+        $options = array_change_key_case($options) ?: [];
+        $options = $options + $defaultOptions ;
+
+        for ($i = 0; $i < $options['maxDepth']; $i++) {
+            $childCategories = $this->getChildrenByParents($parentIDs, $options['permission']);
```

Comment:
This makes a request to the DB every time it's called.

Reply:
added caching